impeller/entity/contents/contents.h

```diff
@@ -83,6 +83,19 @@ class Contents {
 
   void SetColorSourceSize(Size size) { color_source_size_ = size; }
 
+  /// @brief Whether or not this contents can accept the opacity peephole
+  ///        optimization.
+  ///
+  ///        By default all contents return false. Contents are responsible
+  ///        for determining whether or not their own geometries intersect in
+  ///        a way that makes accepting opacity impossible. It is always safe
+  ///        to return false, especially if computing overlap would be
+  ///        computationally expensive.
+  virtual bool CanAcceptOpacity(const Entity& entity) const;
```

Comment:
Can this just be a single call. Say "virtual bool InheritOpacity(...)". The return value  can indicate if the operation succeeded.

Reply:
We have to make sure the entities in a pass aren't overlapping too, so we can't combine them. basically there would need to be two levels of checks:

1. For each entity in the pass, it needs to be able to accept opacity and not overlap other entities.
2. WIth an entities contents, its geometry needs to be non-overlapping.

Reply:
sgtm